pkg/storage/utils/decomposedfs/decomposedfs.go

```diff
@@ -276,11 +276,23 @@ func (fs *Decomposedfs) Postprocessing(ch <-chan events.Event) {
 				failed = true
 			}
 
+			getParent := func() *node.Node {
+				p, err := node.ReadNode(ctx, fs.lu, up.Info.Storage["SpaceRoot"], n.ParentID, false, nil, true)
+				if err != nil {
+					log.Error().Err(err).Str("uploadID", ev.UploadID).Msg("could not read parent")
+					return nil
+				}
+				return p
+			}
+
 			now := time.Now()
-			if p, err := node.ReadNode(ctx, fs.lu, up.Info.Storage["SpaceRoot"], n.ParentID, false, nil, true); err != nil {
-				log.Error().Err(err).Str("uploadID", ev.UploadID).Msg("could not read parent")
-			} else {
-				// update parent tmtime to propagate etag change
+			if failed {
+				// propagate sizeDiff after failed postprocessing
+				if err := fs.tp.Propagate(ctx, up.Node, -up.SizeDiff); err != nil {
+					log.Error().Err(err).Str("uploadID", ev.UploadID).Msg("could not propagate tree size change")
+				}
+			} else if p := getParent(); p != nil {
```

Comment:
I used the `getParent` func to log the error in case it can't get the parent. But we could probably do `n.Parent` instead of `node.ReadNode` in line `280` if that is what you mean?